src/Cache/RedisDistributedCache.cs

```diff
@@ -153,22 +172,21 @@ public void Set<T>(string key, T value, DistributedCacheEntryOptions options)
 
             using (var client = this.redisManager.GetClient())
             {
-                var entry = new RedisDistributedCacheEntry<T>
-                            {
-                                    Value = value,
-                                    AbsoluteExpiration = absoluteExpiration?.Ticks,
-                                    SlidingExpiration = options.SlidingExpiration?.Ticks
-                            };
+                var typedClient = client.As<T>();
+                var redisHash = typedClient.GetHash<string>(key);
+                
+                var typedLongClient = client.As<long>();
+                var redisLongHash = typedLongClient.GetHash<string>(key);
+
+                typedClient.SetEntryInHash(redisHash, DataKey, value);
+                typedLongClient.SetEntryInHash(redisLongHash, AbsoluteExpirationKey, absoluteExpiration?.Ticks ?? NotPresent);
+                typedLongClient.SetEntryInHash(redisLongHash, SlidingExpirationKey, options.SlidingExpiration?.Ticks ?? NotPresent);
 
                 var expiresIn = this.GetExpirationInSeconds(creationTime, absoluteExpiration, options);
 
                 if (expiresIn.HasValue)
                 {
-                    client.Set(key, entry, TimeSpan.FromSeconds(expiresIn.Value));
-                }
-                else
-                {
-                    client.Set(key, entry);
+                    typedClient.ExpireEntryIn(key, TimeSpan.FromSeconds(expiresIn.Value));
```

Comment:
is expiresIn.Value correct? if (expiresIn.HasValue) == false else ...

Reply:
Yes. `expiresIn` is a nullable number of seconds for initial expiration